Source/Core/InputCommon/ControllerEmu/ControlGroup/Cursor.cpp

```diff
@@ -69,80 +77,92 @@ Cursor::ReshapeData Cursor::GetReshapableState(bool adjusted)
   const ControlState y = controls[0]->GetState() - controls[1]->GetState();
   const ControlState x = controls[3]->GetState() - controls[2]->GetState();
 
-  // Return raw values. (used in UI)
+  // Return raw values
   if (!adjusted)
     return {x, y};
 
-  return Reshape(x, y, 0.0);
+  // Values are clamped later on, the maximum movement between two frames
+  // should not be clamped in relative mode
+  return Reshape(x, y, 0.0, std::numeric_limits<ControlState>::infinity());
 }
 
 ControlState Cursor::GetGateRadiusAtAngle(double ang) const
 {
   return SquareStickGate(1.0).GetRadiusAtAngle(ang);
 }
 
-Cursor::StateData Cursor::GetState(const bool adjusted)
+Cursor::StateData Cursor::GetState(float absolute_time_elapsed, bool is_ui)
 {
-  if (!adjusted)
-  {
-    const auto raw_input = GetReshapableState(false);
-
-    return {raw_input.x, raw_input.y};
-  }
-
   const auto input = GetReshapableState(true);
 
-  // TODO: Using system time is ugly.
+  int i = is_ui ? 1 : 0;
+
   // Kill this after state is moved into wiimote rather than this class.
   const auto now = Clock::now();
   const auto ms_since_update =
-      std::chrono::duration_cast<std::chrono::milliseconds>(now - m_last_update).count();
-  m_last_update = now;
-
-  const double max_step = STEP_PER_SEC / 1000.0 * ms_since_update;
+      std::chrono::duration_cast<std::chrono::microseconds>(now - m_last_update[i]).count() /
```

Comment:
```
// Include <ratio>
using milli_with_remainder = std::chrono::duration<double, std::milli>;

const auto ms_since_update = std::chrono::duration_cast<milli_with_remainder>(now - m_last_update[i]).count();
```

I can't find the link to the SO post but I believe that is more accurate

Reply:
Done that in: https://github.com/dolphin-emu/dolphin/pull/9489, will either backport it here or merge it in a new PR, depending how we proceed.